tests/unit/GraphAlgorithms.cpp

```diff
@@ -178,9 +111,43 @@ BOOST_AUTO_TEST_CASE(GetBackedgesTest) {
   revng_check(Target == LG.Entry);
 
   // Check the reachability set described by the only backedge present.
-  BlockSet Reachables = nodesBetween(Target, Source);
+  BlockSet Reachables = nodesBetweenNew(Target, Source);
   revng_check(Reachables.size() == 2);
   revng_check(Reachables.contains(LG.Entry));
   revng_check(Reachables.contains(LG.LoopLatch));
   revng_check(LG.Entry != LG.LoopLatch);
+  revng_check(Reachables[0] == LG.LoopLatch);
+  revng_check(Reachables[1] == LG.Entry);
+}
+
+BOOST_AUTO_TEST_CASE(NestedRegionTest) {
+  // Create the graph.
+  using NodeType = BidirectionalNode<MyBidirectionalNode>;
+  auto NLG = createNLGGraph<NodeType>();
+  using EdgeDescriptor = dt::EdgeDescriptor<NodeType *>;
+  using EdgeSet = llvm::SmallSetVector<EdgeDescriptor, 4>;
+  using BlockSet = llvm::SmallSetVector<NodeType *, 4>;
+  using BlockSetVect = llvm::SmallVector<BlockSet>;
+
+  // Compute the backedges set.
+  EdgeSet Backedges = getBackedges(NLG.Entry);
+  revng_check(Backedges.size() == 2);
+  revng_check(Backedges[0].first == NLG.LoopLatch);
+  revng_check(Backedges[0].second == NLG.SecondLoopHeader);
+  revng_check(Backedges[1].first == NLG.SecondLoopLatch);
+  revng_check(Backedges[1].second == NLG.LoopHeader);
+
+  BlockSetVect Regions;
+  for (EdgeDescriptor Backedge : Backedges) {
+    BlockSet RegionNodes = nodesBetweenNew(Backedge.second, Backedge.first);
+    Regions.push_back(std::move(RegionNodes));
+  }
+
+  revng_check(Regions.size() == 2);
+  revng_check(Regions[0][0] == NLG.LoopLatch);
+  revng_check(Regions[0][1] == NLG.SecondLoopHeader);
+  revng_check(Regions[1][0] == NLG.SecondLoopLatch);
+  revng_check(Regions[1][1] == NLG.LoopHeader);
+  revng_check(Regions[1][2] == NLG.SecondLoopHeader);
+  revng_check(Regions[1][3] == NLG.LoopLatch);
```

Comment:
* Let's not use the term regions
* Add a comment saying that we want to test that nodes reachable from the extremes of the a backedge is basically the whole loop

Reply:
In b39eb194c7c8c536266997effafe3b5ec39dd6bb I removed the _region_ terminology, going for a more neutral _loop_, since here we do not have the _region_ concept.

I also added a comment explaining what's the purpose of the above tests.